### ISSUE TITLE
False positive for custom selectors in `selector-pseudo-class-no-unknown`

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#2141]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

False positive for custom selectors in `selector-pseudo-class-no-unknown`.

> Which rule, if any, is this issue related to?

`selector-pseudo-class-no-unknown`

> What CSS is needed to reproduce this issue?

e.g.

```css
:--heading { 
  margin-top: 0 
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-pseudo-class-no-unknown": true
  }
}
```

> Which version of stylelint are you using?

`7.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

`postcss-cssnext`

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
1:1    Unexpected unknown pseudo-class selector ":-heading" (selector-pseudo-class-no-unknown)
```

No time fixed this today.
